src/components/templates/Camp/hooks/useKeyPress.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+
+import { MOVEMENT_INTERVAL } from "./useKeyboardControls";
+
+// TODO: We can probably replace this with the mousetrap library
+//   https://craig.is/killing/mice
+export const useKeyPress = function (targetKey: string) {
```

Comment:
[TODO] see if mousetrap works better for this

- https://craig.is/killing/mice

Alternatively, using `useRef()` rather than `useState()` here may allow us to not break memoisation so often, but then we may need an alternative method of telling react to re-render when we actually want it to.